### PR TITLE
Rename FileRevision to BlobRevision

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/pd6fOgwf/800-follow-up-steps-on-minimal-workflow-for-inline-attachments

Previously we used FileRevision to encapsulate the versioning of the
bytes ('blob') associated with an image or a file attachment. This
renames FileRevision to BlobRevision to reduce the reuse of the word
'file' throughout the codebase, as the repetition was confusing.

This also updates the debug view for an edition so that it's
automatically available to devs and shows any file attachments.

### CODE REVIEW

File: docs/adr/0009-modelling-history.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # 9. Modelling history
 
-Date: 2018-12-17  
+Date: 2018-12-17
```
Comment (kevindew) [inline docs/adr/0009-modelling-history.md]: Btw you've lost the new line here

File: docs/adr/0009-modelling-history.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # 9. Modelling history
```
Comment (kevindew) [inline docs/adr/0009-modelling-history.md]: I don't think we should be changing the ADR for this, it represents a document at a point in time rather than a living document.

File: docs/adr/0009/revision-diagram.plantuml [hunk 3]
```diff
@@ -20,7 +20,7 @@ class MetadataRevision {
 }
 
 class ImageRevision {
-  image_file_revision
+  image_blob_revision
```
Comment (kevindew) [inline docs/adr/0009/revision-diagram.plantuml]: Note you'd also have to regenerate the png

File: db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class RenameFileRevisionToBlobRevision < ActiveRecord::Migration[5.2]
```
Comment (kevindew) [inline db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb]: I'd suggest that we do this migration in a separate PR and do the first deploy of the code with it using the old table names but new model names. 

Then we can do another one that flips things over and can break and would be a much smaller amount of changes.

File: db/schema.rb [hunk 5]
```diff
@@ -90,24 +90,24 @@
...
 
-  create_table "file_attachment_file_revisions", force: :cascade do |t|
+  create_table "file_attachment_blob_revisions", force: :cascade do |t|
     t.bigint "blob_id", null: false
     t.bigint "created_by_id"
     t.string "filename", null: false
     t.datetime "created_at", null: false
     t.bigint "size", null: false
-    t.index ["blob_id"], name: "index_file_attachment_file_revisions_on_blob_id"
+    t.index ["blob_id"], name: "index_file_attachment_blob_revisions_on_blob_id"
```
Comment (kevindew) [inline db/schema.rb]: it doesn't look like there are migrations that rename this, and any other, indexes?

File: db/seeds.rb [hunk 6]
```diff
@@ -4,6 +4,6 @@
 
 gds_organisation_content_id = "af07d5a5-df63-4ddc-9383-6a666845ebe9"
 
-user.update!(permissions: [User::PRE_RELEASE_FEATURES_PERMISSION],
+user.update!(permissions: [User::PRE_RELEASE_FEATURES_PERMISSION, User::DEBUG_PERMISSION],
```
Comment (kevindew) [inline db/seeds.rb]: I'm a bit concerned about the test user having debug permission by default. Could we do that if you're in a dev environment?

Could this be a separate commit too?

File: app/controllers/debug_controller.rb [hunk 7]
```diff
@@ -48,17 +48,30 @@ def revision_hash(revision)
...
+  end
+
+  def file_attachment_revision_hash(file_attachment_revision)
+    return nil unless file_attachment_revision
+
+    common_except = %i[id created_at created_by_id]
+    blob_revision = file_attachment_revision.blob_revision.as_json(except: common_except)
+    metadata_revision = file_attachment_revision.metadata_revision.as_json(except: common_except)
+
+    blob_revision.merge(metadata_revision)
```
Comment (kevindew) [inline app/controllers/debug_controller.rb]: Thanks for getting file attachment stuff 👍 

I would suggest also a separate commit for this bit

File: docs/adr/0009-modelling-history.md (hunk 2 above)
Comment (benthorner) [reply docs/adr/0009-modelling-history.md]: I did wonder - undone.

File: db/seeds.rb (hunk 6 above)
Comment (benthorner) [reply db/seeds.rb]: Fair enough - done :-).

File: db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb (hunk 4 above)
Comment (benthorner) [reply db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb]: I had a quick think about possible approaches, but as it quickly became apparent the change is a breaking one, it seemed more valuable to keep everything in one PR to aid understanding. As changing table names should be fast, is there really much value in separating it?

File: db/schema.rb (hunk 5 above)
Comment (benthorner) [reply db/schema.rb]: I doubted myself there for a moment, but thankfully this is done as part of renaming the column.

```
psql content_publisher_development -c '\di' | grep file_revision
```

File: db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb (hunk 4 above)
Comment (kevindew) [reply db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb]: Yup there is a breaking part of it whatever happens. What I'm encouraging is minimising the amount that is in the breaking piece of work.

Thus you'd have as much as possible in something that can be merged in as normal and avoids any special deployment considerations and is pretty safe to undo. Then you'd have a change that is db migration and model changes that needs to go from merge to deploy as quick as possible and is rather painful to undo.

File: db/schema.rb (hunk 5 above)
Comment (kevindew) [reply db/schema.rb]: Oh TIL didn't realise the rename commands had extra side effects

File: db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb (hunk 4 above)
Comment (benthorner) [reply db/migrate/20190424145213_rename_file_revision_to_blob_revision.rb]: How about now? (I didn't think this was doable for the column renames initially.)
